File: LENTABP10/src/main/java/com/lenta/lentabp10/models/repositories/ITaskExciseStampRepository.java
package com.lenta.lentabp10.models.repositories;

import com.lenta.lentabp10.models.task.TaskExciseStamp;
import com.lenta.shared.models.core.IProduct;

import java.util.List;

public interface ITaskExciseStampRepository {
    List<TaskExciseStamp> findExciseStampsOfProduct(IProduct product);

    boolean addExciseStamp(TaskExciseStamp exciseStamp);
    boolean deleteExciseStamp(TaskExciseStamp exciseStamp);
    boolean deleteExciseStampsForProduct(IProduct product);
    boolean addExciseStamps(List<TaskExciseStamp> exciseStamps150);
    boolean deleteExciseStamps(List<TaskExciseStamp> exciseStamps150);
    void clear();

    TaskExciseStamp get(int index);
    int lenght();
}

File: writeoff_breakage/src/main/java/com/lenta/bp10/models/repositories/ITaskExciseStampRepository.kt
package com.lenta.bp10.models.repositories

import com.lenta.bp10.models.task.TaskExciseStamp
import com.lenta.shared.models.core.ProductInfo

interface ITaskExciseStampRepository {
    fun getExciseStamps(): List<TaskExciseStamp>
    fun findExciseStampsOfProduct(product: ProductInfo): List<TaskExciseStamp>
    fun addExciseStamp(exciseStamp: TaskExciseStamp): Boolean
    fun deleteExciseStamp(exciseStamp: TaskExciseStamp): Boolean
    fun deleteExciseStampsForProduct(product: ProductInfo): Boolean
    fun addExciseStamps(exciseStamps150: List<TaskExciseStamp>): Boolean
    fun deleteExciseStamps(exciseStamps150: List<TaskExciseStamp>): Boolean
    fun clear()
    fun get(index: Int): TaskExciseStamp
    fun lenght(): Int
}